src/harness/testcases/WINNF_FT_S_SIQ_testcase.py

```diff
@@ -206,6 +206,154 @@ def test_WINNF_FT_S_SIQ_8(self):
+
+    # Update PPA record with device_a's CBSD ID and Inject data
+    ppa_record1['ppaInfo']['cbsdReferenceId'] = [cbsd_ids[0]]
+    self._sas_admin.InjectPalDatabaseRecord(pal_record1[0])
+    zone_id = self._sas_admin.InjectZoneData({'record': ppa_record1})
+    self.assertTrue(zone_id)
+
+    # Update PPA record with device_c's CBSD ID and Inject data
+    ppa_record2['ppaInfo']['cbsdReferenceId'] = [cbsd_ids[1]]
+    self._sas_admin.InjectPalDatabaseRecord(pal_record2[0])
+    zone_id = self._sas_admin.InjectZoneData({'record': ppa_record2})
+    self.assertTrue(zone_id)
+
+    # Create Spectrum Inquiry requests
+    spectrum_inquiry_0 = {
+        'cbsdId': cbsd_ids[0],
+        'inquiredSpectrum': [{
+            'lowFrequency': 3550000000,
+            'highFrequency': 3700000000
+        }]
+    }
+    spectrum_inquiry_1 = {
+        'cbsdId': cbsd_ids[1],
+        'inquiredSpectrum': [{
+            'lowFrequency': 3600000000,
+            'highFrequency': 3700000000
+        }]
+    }
+    spectrum_inquiry_2 = {
+        'cbsdId': cbsd_ids[2],
+        'inquiredSpectrum': [{
+            'lowFrequency': 3550000000,
+            'highFrequency': 3700000000
+        }]
+    }
+    spectrum_inquiry_3 = {
+        'cbsdId': cbsd_ids[3],
+        'inquiredSpectrum': [{
+            'lowFrequency': 3550000000,
+            'highFrequency': 3700000000
+        }]
+    }
+    request = {
+        'spectrumInquiryRequest': [
+            spectrum_inquiry_0, spectrum_inquiry_1, spectrum_inquiry_2,
+            spectrum_inquiry_3
+        ]
+    }
+
+    # Check Spectrum Inquiry response
+    response = self._sas.SpectrumInquiry(request)['spectrumInquiryResponse']
+    for resp_num, resp in enumerate(response):
+      self.assertEqual(resp['cbsdId'], cbsd_ids[resp_num])
+      self.assertEqual(resp['response']['responseCode'], 0)
+
+    # Response 0 and 1 contain at least 1 PAL channelType in availableChannel.
+    for resp_num in [0, 1]:
+      self.assertTrue(
+          any(channel['channelType'] == 'PAL'
+              for channel in response[resp_num]['availableChannel']))
+      self.assertTrue(
+          all(channel['ruleApplied'] == 'FCC_PART_96'
+              for channel in response[resp_num]['availableChannel']))
+      # Verify that the upper frequency is valid (lower frequency will be
+      # checked below).
+      self.assertTrue(
+          all(channel['frequencyRange']['highFrequency'] <= 3700000000
+              for channel in response[resp_num]['availableChannel']))
+    # Check the frequency range of assigned PAL and GAA channels.
+    for channel in response[0]['availableChannel']:
+      if channel['channelType'] == 'PAL':
+        self.assertTrue(
+            channel['frequencyRange']['lowFrequency'] == 3550000000)
+        self.assertTrue(
+            channel['frequencyRange']['highFrequency'] == 3560000000)
+      if channel['channelType'] == 'GAA':
```

Comment:
if we deleted the first loop, this can be "else" with assert that ['channelType'] == 'GAA', right?

Reply:
Done.

Reply:
thank you very much for your modifications, they seem ok, I have one additional suggestion ,if you think it is ok, instead of the values of frequencies(3550000000, 3560000000, 3600000000, 3610000000 ) of PALs, we could use the variables that we have already pal_low_frequency1, pal_high_frequency1, pal_low_frequency2, frequency2,  right?

Reply:
Sure, that makes sense. I have changed the way you suggested. Could you please take a look, thanks.